lib/syskit/test/profile_assertions.rb

```diff
@@ -15,10 +15,15 @@ module ProfileAssertions
             class ProfileAssertionFailed < Roby::ExceptionBase
                 attr_reader :actions
 
-                def initialize(assertion_name, actions, original_error)
+                def initialize(assertion_name, actions, original_errors)
                     @assertion_name = assertion_name
                     @actions = Array(actions)
-                    super([original_error])
+                    # When capture errors is active, its possible to receive an array of
+                    # exceptions instead of a single one.
+                    unless original_errors.kind_of? Array
+                        original_errors = [original_errors]
+                    end
```

Comment:
FYI, `Array()` is the Ruby canonical conversion method, that converts anything-to-array (but keeps arrays as-is), i.e.

```
original_errors = Array(original_errors)
```